### ISSUE TITLE
Feedback and Suggestions

### ISSUE THREAD

**Issue by Camawama** [Povstalec/StellarView#2]
This is absolutely amazing! 100% will be using it in all my future modpacks. I was going to create a mod exactly like this with the planned random events and everything so this is seriously amazing to see it done!!

here are the only bugs I've found (so far):

#1 Shader support. I cannot get complimentary shaders to render the mod stars.

**Comment by Camawama** [comment]
Also had the idea to fade the stars into the sky like in this video from snapchat:

https://t.snapchat.com/FuB7hK6T

**Comment by Camawama** [comment]
Support for EnhancedCelestials mod (Blood Moon, Harvest Moon, Blue Moon)

**Comment by Noobly-Walker** [comment]
Even better: Add eclipse support. The moon already passes in front of the sun, you might as well commit at this point.

**Comment by Camawama** [comment]
Another Idea: Different colored meteors! My brother and I watched meteor showers and saw different colors like green in the past. Would be cool to see added!

**Comment by Povstalec** [comment]
> #1 Shader support. I cannot get complimentary shaders to render the mod stars. 
> 
I would like to figure that one out, but unfortunately, my knowledge of how shaders work is very limited, so I don't know how to do it yet.

> #2 More Twinkle to Stars
> 
Can you elaborate on what you mean by twinkle?

> #3 Meteor events (fragments fall onto Earth's surface)
> 
What exactly do you mean by fragments? A visual of the meteor blowing up in the sky, or a physical fragment that falls down and you can find? I'd like to keep the mod Client-only, so a physical fragment that would require accessing the Server side is not on the table.

**Comment by Povstalec** [comment]
> Also had the idea to fade the stars into the sky like in this video from snapchat:
> 
> https://t.snapchat.com/FuB7hK6T
> 
It does look pretty cool, although I'm not sure if it would go well with the Minecraft-y aesthetic I'm going for

**Comment by Povstalec** [comment]
> Support for EnhancedCelestials mod (Blood Moon, Harvest Moon, Blue Moon)
>
Yes, support for Enhanced Celestials is something I want to add eventually

**Comment by Povstalec** [comment]
> Even better: Add eclipses outright. The moon already passes in front of the sun, you might as well commit at this point.

That's actually a pretty neat idea, although I'm not sure how well changing the darkness of the world would go with other mods if they were installed

**Comment by Povstalec** [comment]
> Another Idea: Different colored meteors! My brother and I watched meteor showers and saw different colors like green in the past. Would be cool to see added!

Colored meteors sound like a cool idea and they would be fairly simple to implement, definitely gonna add them

**Comment by Noobly-Walker** [comment]
> Can you elaborate on what you mean by twinkle?
Various masses of high and low pressure air moving around in the atmosphere causes the refraction of starlight to shift a bit, causing some of them to flicker a bit.

> That's actually a pretty neat idea, although I'm not sure how well changing the darkness of the world would go with other mods if they were installed
That's the neat part. You don't. You can dim the skybox briefly, but you don't have to dim it completely, only so much as a few of the brighter stars/planets become faintly visible. Besides, I don't think it would be appreciated much if, every "month", for a few ticks mobs were spawning in the middle of the day. That does remind me however, that the moon phases need to be offset by... half a day? The moon starts covering the sun as a waning crescent, and once the eclipse totality is reached, only then does it transition to a new moon phase.

**Comment by Noobly-Walker** [comment]
If changing the sky brightness is specifically what you were talking about, then maybe make it optional? I don't know much about mod compatibility.

**Comment by Povstalec** [comment]
Hmm, I suppose dimming the skybox alone without affecting the lighting is doable, so sure

Oh and thanks for informing me about the moon phase needing to be offset

**Comment by Camawama** [comment]
Leaving the mod client-side helps a lot with compatibility and such but just imagine how cool it would be to have meteor fragments crash from the sky! If you are 100% sure you wish to keep it client-side only, a cool little visual would be fun! Maybe you see a shooting star but it gets larger and brighter before splitting into fragments that dissipate in the atmosphere. 

p.s. I am super happy to see more conversation here regarding my ideas. I didn't check it for a while!

**Comment by Noobly-Walker** [comment]
I can foresee a server-side version of the mod. While optional, it would allow everyone's viewing experience of the sky to be synchronized. Everyone would see that supernova, or that cool planetary alignment, or that meteor shower. The server could also, using resource packs and data packs, set up a custom solar system that everyone would be able to see.
Might require some network BS to pull off though.

**Comment by Camawama** [comment]
> I can foresee a server-side version of the mod. While optional, it would allow everyone's viewing experience of the sky to be synchronized. Everyone would see that supernova, or that cool planetary alignment, or that meteor shower. The server could also, using resource packs and data packs, set up a custom solar system that everyone would be able to see. Might require some network BS to pull off though.
> 
> PS: I rarely have anything better to do as of late.
> 
> PPS: I can't help but feel this could've been multiple issues or something.

That sounds like it would be nice!

**Comment by Povstalec** [comment]
> I can foresee a server-side version of the mod. While optional, it would allow everyone's viewing experience of the sky to be synchronized. Everyone would see that supernova, or that cool planetary alignment, or that meteor shower.

Realistically speaking, the view of the sky is synchronized already, since it's only based on the game time and nothing else. 

> The server could also, using resource packs and data packs, set up a custom solar system that everyone would be able to see.

Currently I've rewritten basically all of the code to make it more intuitive to use and have objects/events that move relative to the galaxy and not the sky, which now now makes it so players can see the same Supernova from 2 different planets. I'm hoping I can figure out a way to make it all work with Resource Packs, which would on its own basically be enough to set up a Solar System for the server.  

Outside of that Server customization, reasons to make it a Server-side mod are few and don't outweigh the pros of keeping it Client-side yet (for example, being able to use it even when playing on Vanilla Servers, which I suppose could be achieved with having a bunch of checks that activate and deactivate parts of the code based on whether or not it's installed on the Server), so I'm still not all too convinced a Server-side version would be needed.

**Comment by Noobly-Walker** [comment]
The server-side mod would be a compatible but unrelated mod. The client mod wouldn't require the server mod, and the server mod wouldn't require the client mod, but with the mod on both, things could be synced. The main assumption is, the sky's seed is different from client to client. If the assumption is unfounded, then it would literally just be for custom planets or something to be synced, I guess.

**Comment by Noobly-Walker** [comment]
I thought the stars were based loosely on the world seed, but alright then.

**Comment by Povstalec** [comment]
> I thought the stars were based loosely on the world seed, but alright then.

Nope, world seed is actually inaccessible on the Client-side, as far as I'm aware. No matter the World seed, the actual sky seed in Vanilla is always 10842L (with L indicating Long type, not actually being a letter)

**Comment by Camawama** [comment]
> > I thought the stars were based loosely on the world seed, but alright then.
> 
> Nope, World seed is actually inaccessible on the Client-side, as far as I'm aware. No matter the World seed, the actual sky seed in Vanilla is always 10842L (with L indicating Long type, not actually being a letter)

Is that due to the obfuscation of the code? Man I really hate that they decided to hide the code.

**Comment by Povstalec** [comment]
> Is that due to the obfuscation of the code? Man I really hate that they decided to hide the code.

I don't know why obfuscation would have anything to do with that. Level consists of ServerLevel and ClientLevel. The Seed is saved on ServerLevel, while Client can only acces the ClientLevel side of things

**Comment by Camawama** [comment]
> > Is that due to the obfuscation of the code? Man I really hate that they decided to hide the code.
> 
> I don't know why obfuscation would have anything to do with that. Level consists of ServerLevel and ClientLevel. The Seed is saved on ServerLevel, while Client can only acces the ClientLevel side of things

Gotcha, I was thinking of something else. I tried making a mod at one point that disables seeds and uses one set seed (for someone's story modpack) but I was running into that issue.

**Comment by Camawama** [comment]
Just stopping by to say I love all the progress so far! Can't wait to see more!

**Comment by Camawama** [comment]
I know you said you were already going to add colored shooting stars but here is a cool reference photo!
![image](https://github.com/Povstalec/StellarView/assets/43620900/fbb2a30c-9e90-4daa-a026-22e245ae2cb8)

**Comment by Camawama** [comment]
![image](https://github.com/Povstalec/StellarView/assets/43620900/e1a6e223-9e2e-4d6d-8289-ccf70c84daaf)

**Comment by Noobly-Walker** [comment]
several of those in the second image look a bit like comets

**Comment by Povstalec** [comment]
hmm, nice reference for colors

**Comment by Camawama** [comment]
> several of those in the second image look a bit like comets

Don't lie, comets would be awesome too 😄

**Comment by Noobly-Walker** [comment]
I'm genuinely surprised comets haven't been suggested yet. I thought they had been.

**Comment by Noobly-Walker** [comment]
Okay, let me summarize:
- [ ] Support for EnhancedCelestials lunar events
- [ ] Some stars vary in brightness, "twinkling" (bonus points if effect is more noticeable towards the horizon)
- [ ] Solar and lunar eclipses - brief celestial events that occur when the sun, moon, and earth are in alignment.
- [ ] Shooting stars/meteors - chromatic streaks occasionally appearing each night, frequency varying from night to night
- [ ] Star variations - stars vary in color (red to blue) and size
- [ ] Comets - appearing more frequently than supernovae, tails pointing away from the sun. They grow for a few days, then shrink for a few days before disappearing.
Here's some new ones:
- [ ] Seasons - The sun and moon's paths across the sky vary over time, moving from the south to north to south again over periods of 365.2422 minecraft days. New settings for what the starting season should be (most people might like it to start as spring or summer, but I prefer it starting in winter), as well as controls for latitude.
- [ ] Irregular Lunar Orbit - The moon's path across the sky is unrelated to the sun's path across the sky, making eclipses rarer. Additionally, the moon appears to grow and shrink over time. This will notably change how solar eclipses work, the moon's distance being the difference between an annular and total eclipse.

**Comment by Camawama** [comment]
> Okay, let me summarize:
> 
> * [x]  Support for EnhancedCelestials lunar events
> * [ ]  Some stars vary in brightness, "twinkling" (bonus points if effect is more noticeable towards the horizon)
> * [ ]  Solar and lunar eclipses - brief celestial events that occur when the sun, moon, and earth are in alignment (bonus points if the skybox dims when the sun is eclipsed)
> * [ ]  Shooting stars/meteors - chromatic streaks occasionally appearing each night, frequency varying from night to night
> * [ ]  Star variations - stars vary in color (red to blue) and size
> * [ ]  Comets - appearing more frequently than supernovae, tails pointing away from the sun. They grow for a few days, then shrink for a few days before disappearing.
>   Here's some new ones:
> * [ ]  Seasons - The sun and moon's paths across the sky vary over time, moving from the south to north to south again over periods of 365.2422 minecraft days. New settings for what the starting season should be (most people might like it to start as spring or summer, but I prefer it starting in winter), as well as controls for latitude.
> * [ ]  Irregular Lunar Orbit - The moon's path across the sky is unrelated to the sun's path across the sky, making eclipses rarer. Additionally, the moon appears to grow and shrink over time. This will notably change how solar eclipses work, the moon's distance being the difference between an annular and total eclipse.
> * [ ]  Realistic Moon Phases - Moon would take 29.53 minecraft days to complete a cycle of phases, instead of 7. This would desync the moon's phase from the effects on mob spawning based on moon phase, which the mod cannot control, but it would make eclipses last over four times longer. This should be optional, disabled by default.

Dude, amazing summary! Thank you 😄

**Comment by Noobly-Walker** [comment]
Eclipses were asked for in #29. "twinkling" isn't really workable at this time. I'm already working on comets. Does the Earth 'wobble' on its axis at all? I don't remember noticing it.

**Comment by Povstalec** [comment]
It will kinda do that on 0.3, yeah.

**Comment by Noobly-Walker** [comment]
pretty sure it can just be closed then?

**Comment by Povstalec** [comment]
No, not yet
